File: vidaplus-backend-tcc/src/main/java/br/com/vidaplus/vidaplus_backend_tcc/entity/Profissional.java
package br.com.vidaplus.vidaplus_backend_tcc.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Profissional {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @OneToOne
    @JoinColumn(name = "user_id")
    @JsonBackReference
    private User user;

    private String nome;
    private String especialidade;
    private String registro;
}

File: vidaplus_app/lib/models/profissional.dart
class Profissional {
  final String id;
  final String nome;
  final String? registro;
  final String? especialidade;

  Profissional({
    required this.id,
    required this.nome,
    this.registro,
    this.especialidade,
  });

  factory Profissional.fromJson(Map<String, dynamic> json) {
    return Profissional(
      id: json['id'],
      nome: json['nome'],
      registro: json['registro'],
      especialidade: json['especialidade'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nome': nome,
      'registro': registro,
      'especialidade': especialidade,
    };
  }
}